data/transactions/application_test.go

```diff
@@ -129,6 +129,81 @@ func TestEncodedAppTxnAllocationBounds(t *testing.T) {
 	}
 }
 
+func TestAppCallVersioningWellFormed(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	preVersion := protocol.ConsensusV40
+	v5 := []byte{0x05}
+
+	cases := []struct {
+		expectedError string
+		cv            protocol.ConsensusVersion // defaults to future if not set
+		ac            ApplicationCallTxnFields
+	}{
+		{
+			cv: preVersion,
+			ac: ApplicationCallTxnFields{
+				ApplicationID: 1,
+				RejectVersion: 0,
+			},
+		},
+		{
+			expectedError: "tx.RejectVersion is not supported",
+			cv:            preVersion,
+			ac: ApplicationCallTxnFields{
+				ApplicationID: 1,
+				RejectVersion: 1,
+			},
+		},
+		{
+			ac: ApplicationCallTxnFields{
+				ApplicationID: 1,
+				RejectVersion: 0,
+			},
+		},
+		{
+			ac: ApplicationCallTxnFields{
+				ApplicationID: 1,
+				RejectVersion: 1,
+			},
+		},
+		{
+			ac: ApplicationCallTxnFields{
+				ApprovalProgram:   v5,
+				ClearStateProgram: v5,
+				RejectVersion:     0,
+			},
+		},
+		{
+			// Should we allow this, it's kind of non-sensical. Why specify a reject version when creating?
```

Comment:
Maybe lean towards rejecting?

Reply:
Ok, adding check in wellFormed, and changing this test case to expected the error.